### DISCUSSION TITLE
Windows very slow to pick up changed host IPv4 link-local

### DISCUSSION THREAD

**Discussion by topher-3d** [christgau/wsdd#134] [Q&A]
Curious if anyone else has experienced this or has any ideas for solutions.

* Device A is running wsdd (v0.6.4 with Python 3.5 on PetaLinux).
* Laptop B is running Windows 10, fully up-to-date.
* Both are on the same wired subnet with no DNS server or gateway. They both acquire link-local IPv4 (and IPv6) addresses.

When I reboot A, with wireshark I see the WSD Bye packet and others, then Hello packets promptly after reboot. Its IPv6LL address is static but its IPv4 address usually changes. So far, so good. Laptop "B" can immediately ping A using IPv6, but "ping A -4" fails for _several minutes_(!!), continuing to use the old address, until some time when I can only guess that the old address expires internally and it switches to the new address that had been announced several minutes prior. I don't see any traffic on port 3702 at the time Windows begins resolving to the new address for A.

What might explain this, or better yet fix it? FWIW, it _feels_ to me like a bug in Windows 10, not wsdd.

**Comment by christgau** [comment]
First of all, this discussion appears to be pretty unrelated to wsdd. The primary task of wsdd is to make hosts appear as icon in the network view of windows explorer. There is only little - if any - connection to pings or how fast windows picks up IP addresses. 

The only relationship might be that when you double click on the icon windows tries to resolves the shown hostname. After that, the resolved mapping between hostname and IP address is usually cached inside windows. From your description, I assume that `ping A -4`  means pinging device A by its hostname (would not make much sense otherwise). When the IP address of device A changes (maybe due to APIPA), Windows very likely will not know about this. Therefore, the cached IP address will be used when pinging the device by its name. You can flush the cache by calling `ipconfig /flushdns` on the command line with administrative privileges. So, no, this is not a Windows bug from my perspective. 

Nevertheless, it appears as if you do not have a DHCP server in your  gateway-less network. But then why you do not chose for setting static IPv4 addresses for the hosts. This would solve the problem as well.
